### PR TITLE
Fix Monitor test

### LINKED ISSUE #2410: TestFileMonitor is flaky
https://travis-ci.org/GoogleContainerTools/skaffold/jobs/554538507

### PR DESCRIPTION
**Pull request by dgageot**
Fixes #2410

Also remove the unused function arguments

### CODE REVIEW

**Comment by dgageot** [comment]
@corneliusweig would you mind taking a look at this PR? I'm trying to fix all known flaky tests

**Comment by dgageot** [comment]
This test was green 10 times with this PR: https://github.com/GoogleContainerTools/skaffold/pull/2414. I'm trying 20 more runs

**Comment by corneliusweig** [comment]
I'm taking a look (can't assign myself).

File: pkg/skaffold/filemon/monitor_test.go [hunk 1]
```diff
@@ -29,76 +26,59 @@ import (
...
-			// Watch folder
 			monitor := NewMonitor()
-			err := monitor.Register(tmpDir.List, folderChanged.call)
-			t.CheckNoError(err)
 
-			// Run the watcher
-			ctx, cancel := context.WithCancel(context.Background())
-			var stopped sync.WaitGroup
-			stopped.Add(1)
-			go func() {
```
Comment (corneliusweig) [inline pkg/skaffold/filemon/monitor_test.go]: Nice, that's much simpler.

It really makes more sense to test synchronous code with a synchronous test :)

**Comment by dgageot** [comment]
Thanks @corneliusweig for reviewing!

**Pull request merged**